src/v/cluster/rm_stm.cc

```diff
@@ -477,25 +477,32 @@ ss::future<tx_errc> rm_stm::do_prepare_tx(
         co_return tx_errc::conflict;
     }
 
-    auto batch = make_prepare_batch(marker);
-    auto reader = model::make_memory_record_batch_reader(std::move(batch));
-    auto r = co_await _c->replicate(
-      etag,
-      std::move(reader),
-      raft::replicate_options(raft::consistency_level::quorum_ack));
+    if (is_transaction_ga()) {
+        // We can skip writeing prepare marker in rm_stm, because all data
+        // batcher is written with quorum_ack, so we do not need marker to be
+        // sure that all replicas replicate and flush data batches
+        apply_prepare(marker);
```

Comment:
Continuing [my other comment](https://github.com/redpanda-data/redpanda/pull/6527/files#r980528884), it looks like you are relying on this RPC to update state here. This defeats the purpose of `acks=all` right? With `acks=all`, I imagined a commit transaction call to be a local decision to the coordinator but we still are making another round of RPCs.

Reply:
Bharath is right, txn coordinator should not do the prepare rpc call at all